tools/run_tests/python_utils/report_utils.py

```diff
@@ -46,8 +46,22 @@ def _filter_msg(msg, output_format):
 def render_junit_xml_report(resultset, xml_report, suite_package='grpc',
                             suite_name='tests'):
   """Generate JUnit-like XML report."""
-  root = ET.Element('testsuites')
-  testsuite = ET.SubElement(root, 'testsuite', id='1', package=suite_package,
+  tree = add_junit_xml_results(resultset, suite_package, suite_name, '1')
+  create_xml_report_file(xml_report, tree)
+
+def create_xml_report_file(xml_report, tree):
+  """Generate JUnit-like report file from xml tree ."""
+  # ensure the report directory exists
+  report_dir = os.path.dirname(os.path.abspath(xml_report))
+  if not os.path.exists(report_dir):
+    os.makedirs(report_dir)
+  tree.write(xml_report, encoding='UTF-8')
+
+def add_junit_xml_results(resultset, suite_package, suite_name, id,
```

Comment:
Just to clarify, "suite_package", "suite_name" and "id" are not ignored.  They become the attribute of the new testsuite element under the old_tree of "testsuites".

See below for example:

```
<testsuites>
<testsuite errors="0" failures="0" id="6474f42c-25ae-4de9-a580-e2300cbeb8f6" name="go__go1.7:master" package="grpc_interop_matrix" timestamp="2017-07-05T11:16:12.256313">
<testcase name="cancel_after_first_response" time="16.6104240417"/>
<testcase name="ping_pong" time="17.6899580956"/>
<testcase name="client_streaming" time="18.3408219814"/>
<testcase name="server_streaming" time="18.156867981"/>
<testcase name="empty_stream" time="13.1786220074"/>
<testcase name="timeout_on_sleeping_server" time="16.0977790356"/>
<testcase name="empty_unary" time="13.9597160816"/>
<testcase name="cancel_after_begin" time="14.9083271027"/>
<testcase name="large_unary" time="17.2520859241"/>
</testsuite>
<testsuite errors="0" failures="0" id="4cb8d3b8-56db-4375-a056-3a955175d16b" name="go__go1.8:master" package="grpc_interop_matrix" timestamp="2017-07-05T11:16:30.792714">
<testcase name="cancel_after_first_response" time="18.0730729103"/>
<testcase name="ping_pong" time="14.6083629131"/>
<testcase name="client_streaming" time="18.5181870461"/>
<testcase name="server_streaming" time="17.1722960472"/>
<testcase name="empty_stream" time="15.1755671501"/>
<testcase name="timeout_on_sleeping_server" time="18.3323328495"/>
<testcase name="empty_unary" time="16.8497140408"/>
<testcase name="cancel_after_begin" time="16.2281429768"/>
<testcase name="large_unary" time="17.6928610802"/>
</testsuite>
</testsuites>
```

Honest, I found it is easier to have a single file to look at for results rather than a bunch of files.  I have made some change to the API hopefully to address your other concerns.